storage-proofs-porep/src/encode.rs

```diff
@@ -20,3 +20,7 @@ pub fn decode<T: Domain>(key: T, value: T) -> T {
     result -= key;
     result.into()
 }
+
+pub fn decode_fr(key: &mut Fr, value: Fr) {
+    *key = value - *key;
```

Comment:
Nitpick (no need to change): For symmetry with `encode_fr` it could be `*key = = value` and it could be called from `decode()` like `encode_fr` is called from `encode`.

Reply:
Yup, makes sense. You mean something like:
```rust
pub fn decode<T: Domain>(key: T, value: T) -> T {
    let value: Fr = value.into();
    let mut key: Fr = key.into();
    decode_fr(&mut key, &value);
    value.into()
}

pub fn decode_fr(key: &mut Fr, value: &Fr) {
    *key = value - key;
}
```